integrations/mixpanel/mixpanel.go

```diff
@@ -1,46 +1,186 @@
+// Page forwards the page-events to Mixpanel
+func (m Mixpanel) Page(page integrations.Page) (err error) {
+	// Mixpanel has no special api for Page events, so just pass on the Page-event to Track
+	// Name is the name of the page
+	pageEvent := integrations.Event{}
+	pageEvent.Name = page.Name
+	pageEvent.UserID = page.UserID
+	pageEvent.Properties = page.Properties
+	pageEvent.Properties["url"] = page.Url
+	pageEvent.Properties["event"] = "page"
+	pageEvent.Timestamp = page.Timestamp
+	pageEvent.ReceivedAt = page.ReceivedAt
+	err = m.Track(pageEvent)
 	return
 }
 
 // Enabled returns wether or not the Mixpanel integration is enabled/configured
 func (Mixpanel) Enabled() bool {
-	return apiKey() != "" && token() != ""
+	return token() != ""
 }
 
-func apiKey() string {
-	return os.Getenv("MIXPANEL_API_KEY")
+func (api mixpanelAPIProduction) request(method string, endpoint string, payload []byte) (err error) {
+	apiUrl := api.Url + endpoint
+	req, err := http.NewRequest(method, apiUrl, nil)
+	// Mixpanel needs the request to be GET http://<api-url>?data=<base64-encoded payload>
+	q := req.URL.Query()
+	logrus.Info(base64.StdEncoding.EncodeToString(payload))
+	q.Add("data", base64.StdEncoding.EncodeToString(payload))
+
+	// Hack, when using the /import-endpoint api_key must be
+	// passed as a query param as well
+	if endpoint == "import" {
+		q.Add("api_key", token())
+	}
+
+	req.URL.RawQuery = q.Encode()
+
+	client := &http.Client{}
+	resp, err := client.Do(req)
+	if resp != nil {
+		defer resp.Body.Close()
+	}
+	if err != nil {
+		logrus.WithError(err).WithField("method", method).WithField("endpoint", endpoint).WithField("payload", string(payload[:])).Error("Error sending request to Mixpanel api")
+		return
+	}
+
+	responseBody, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		logrus.WithError(err).WithField("method", method).WithField("endpoint", endpoint).WithField("payload", string(payload[:])).Error("Error reading body in Mixpanel response")
+		return err
+	}
+
+	// Mixpanel returns a 200OK with a body == 0 when things go wrong
+	if resp.StatusCode != http.StatusOK || string(responseBody) == "0" {
+		logrus.WithField("method", method).WithField("endpoint", endpoint).WithField("payload", string(payload[:])).WithFields(
+			logrus.Fields{
+				"response":    string(responseBody),
+				"HTTP-status": resp.StatusCode}).Error("Mixpanel api returned errors")
+	}
+
+	logrus.WithField("method", method).WithField("endpoint", endpoint).WithField("payload", string(payload[:])).WithFields(
+		logrus.Fields{
+			"response":    string(responseBody),
+			"HTTP-status": resp.StatusCode}).Info("Sent stuff to Mixpanel ", string(req.URL.String()))
+
+	return
 }
 
 func token() string {
 	return os.Getenv("MIXPANEL_TOKEN")
 }
 
 func init() {
-	integrations.RegisterIntegration("mixpanel", Mixpanel{})
+	mixpanel := Mixpanel{}
+	mixpanel.api = &mixpanelAPIProduction{Url: "http://api.mixpanel.com/"}
```

Comment:
Did you try HTTPS like suggested in https://mixpanel.com/help/questions/articles/how-can-i-send-data-to-mixpanel-over-https-instead-of-http-using-the-javascript-library ? Worth trying IMHO as HTTP is worrying for that kind of event :/